lib/src/slices/data_bkg/data_bkg_service.dart

```diff
@@ -208,4 +209,29 @@ class DataBkgService extends ChangeNotifier {
         "." +
         periodSplit[periodSplit.length - 1];
   }
+
+  String _gmailBuildQuery(int appDataGmailLastFetch, String category) {
+    StringBuffer queryBuffer = new StringBuffer();
+    if (appDataGmailLastFetch > 0) {
+      DateTime gmailLastFetch =
+          DateTime.fromMillisecondsSinceEpoch(appDataGmailLastFetch);
+      int secondsSinceEpoch =
+          (gmailLastFetch.millisecondsSinceEpoch / 1000).floor();
+      queryBuffer.write("after:" + secondsSinceEpoch.toString());
+    }
+    if (category != 'category:' && category.isNotEmpty) {
+      if (queryBuffer.isNotEmpty) {
+        queryBuffer.write(" AND ");
+      }
+      queryBuffer.write("label: $category");
+    } else {
+      model.gmailCategoryList.forEach((cat) {
+        if (queryBuffer.isNotEmpty) {
+          queryBuffer.write(" AND ");
+        }
+        queryBuffer.write("NOT $cat");
```

Comment:
is $cat right? shouldn't it be $category

also you don't need to use the "label" tag. gmail supports search by just "category"  https://support.google.com/mail/answer/7190?hl=en